From e7821e0ade3c5815ec254fb69eedf7eac02769cd Mon Sep 17 00:00:00 2001
From: Marek Belisko <marek.belisko@gmail.com>
Date: Tue, 24 Sep 2024 11:19:19 +0200
Subject: [PATCH] u-boot-imx: Take patch from meta-freescale to fix build issue

meta-freescale commit: 161f1b3e69a3cf011a50e9b742fb8c46d61e41e8

Fix following:

2024-09-23T11:17:50.1857749Z 2024-09-23 11:17:50 - INFO     - NOTE: Running task 4514 of 6247 (/__w/meta-pantavisor/meta-pantavisor/build/../poky/meta/recipes-support/libnl/libnl_3.9.0.bb:do_compile_ptest_base)
2024-09-23T11:17:50.6198475Z 2024-09-23 11:17:50 - ERROR    - ERROR: u-boot-imx-2024.04-r0 do_deploy: ExecutionError('/__w/meta-pantavisor/meta-pantavisor/build/tmp-scarthgap/work/imx8qxpc0mek-poky-linux/u-boot-imx/2024.04/temp/run.do_deploy.808', 1, None, None)
2024-09-23T11:17:50.6208454Z 2024-09-23 11:17:50 - ERROR    - ERROR: Logfile of failure stored in: /__w/meta-pantavisor/meta-pantavisor/build/tmp-scarthgap/work/imx8qxpc0mek-poky-linux/u-boot-imx/2024.04/temp/log.do_deploy.808
2024-09-23T11:17:50.6281225Z 2024-09-23 11:17:50 - INFO     - Log data follows:
2024-09-23T11:17:50.6321435Z 2024-09-23 11:17:50 - INFO     - | DEBUG: Executing python function sstate_task_prefunc
2024-09-23T11:17:50.6389186Z 2024-09-23 11:17:50 - INFO     - | DEBUG: Python function sstate_task_prefunc finished
2024-09-23T11:17:50.6442398Z 2024-09-23 11:17:50 - INFO     - | DEBUG: Executing shell function do_deploy
2024-09-23T11:17:50.6519721Z 2024-09-23 11:17:50 - INFO     - | cp: cannot stat '/__w/meta-pantavisor/meta-pantavisor/build/tmp-scarthgap/work/imx8qxpc0mek-poky-linux/u-boot-imx/2024.04/deploy-u-boot-imx/imx-boot': No such file or directory
2024-09-23T11:17:50.6524873Z 2024-09-23 11:17:50 - INFO     - | WARNING: exit code 1 from a shell command.
2024-09-23T11:17:50.6527802Z 2024-09-23 11:17:50 - INFO     - NOTE: recipe u-boot-imx-2024.04-r0: task do_deploy: Failed
2024-09-23T11:17:50.6603605Z 2024-09-23 11:17:50 - ERROR    - ERROR: Task (/__w/meta-pantavisor/meta-pantavisor/build/../meta-imx/meta-imx-bsp/recipes-bsp/u-boot/u-boot-imx_2024.04.bb:do_deploy) failed with exit code '1'

Signed-off-by: Marek Belisko <marek.belisko@gmail.com>
---
 .../recipes-bsp/u-boot/u-boot-imx_2024.04.bb        | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/meta-imx-bsp/recipes-bsp/u-boot/u-boot-imx_2024.04.bb b/meta-imx-bsp/recipes-bsp/u-boot/u-boot-imx_2024.04.bb
index c30af5fe80..03535a63a7 100644
--- a/meta-imx-bsp/recipes-bsp/u-boot/u-boot-imx_2024.04.bb
+++ b/meta-imx-bsp/recipes-bsp/u-boot/u-boot-imx_2024.04.bb
@@ -23,15 +23,10 @@ PROVIDES += "u-boot"
 
 inherit uuu_bootloader_tag
 
-UUU_BOOTLOADER                        = ""
-UUU_BOOTLOADER:mx6-generic-bsp        = "${UBOOT_BINARY}"
-UUU_BOOTLOADER:mx7-generic-bsp        = "${UBOOT_BINARY}"
-UUU_BOOTLOADER_TAGGED                 = ""
-UUU_BOOTLOADER_TAGGED:mx6-generic-bsp = "u-boot-tagged.${UBOOT_SUFFIX}"
-UUU_BOOTLOADER_TAGGED:mx7-generic-bsp = "u-boot-tagged.${UBOOT_SUFFIX}"
-UUU_BOOTLOADER_UNTAGGED                 = ""
-UUU_BOOTLOADER_UNTAGGED:mx6-generic-bsp = "u-boot-untagged.${UBOOT_SUFFIX}"
-UUU_BOOTLOADER_UNTAGGED:mx7-generic-bsp = "u-boot-untagged.${UBOOT_SUFFIX}"
+# The UUU tag goes on the boot partition. For 8+, the boot partition image
+# is imx-boot, so disable UUU-tagging here
+UUU_BOOTLOADER:mx8-generic-bsp = ""
+UUU_BOOTLOADER:mx9-generic-bsp = ""
 
 do_deploy:append:mx8m-generic-bsp() {
     # Deploy u-boot-nodtb.bin and fsl-imx8m*-XX.dtb for mkimage to generate boot binary
-- 
2.25.1

